addons/web/controllers/main.py

```diff
@@ -1585,6 +1586,49 @@ def get_app_url(self, req, app):
         sakey = Session().save_session_action(action)
         return werkzeug.utils.redirect('/web#sa={0}'.format(sakey))
 
+class AttachmentController(http.Controller):
+
+    @http.route('/attachment/download_pdfs', type='http', auth='user')
+    def merge_and_download_pdfs(self, attachment_ids, **data):
+        def _close_streams(streams):
+            for stream in streams:
+                try:
+                    stream.close()
+                except Exception:
+                    pass
+
+        ids = [int(id) for id in attachment_ids.split(',')]
+        original_attachments = request.env['ir.attachment'].browse(ids)
+
+        streams = []
+        readers = []
+        for attachment in original_attachments:
+            if (attachment.name.endswith(".pdf") and attachment.datas):
+                content = base64.decodestring(attachment.datas)
+                try:
+                    stream = io.BytesIO(content)
+                    streams.append(stream)
+                    readers.append(PyPDF2.PdfFileReader(stream))
+                except PyPDF2.utils.PdfReadError:
+                    pass
+
+        if len(readers) == 0:
+            _close_streams(streams)
+            return request.make_response("<script>window.close()</script>")
```

Comment:
instead of returning some javascript, it might be better to raise a userError in the def action_print_original_invoice(self)

Also the mechanism to merge pdf already exists: there is a method called 'merge_pdf'
To prevent writting another controller, another solution might be possible (need to be investigated).
You can look at the following method 'render_qweb_pdf' 'retrieve_attachment' '_post_pdf'